src/interface.jl

```diff
@@ -103,7 +103,9 @@ function transitions(
     N::Integer;
     kwargs...
 )
-    return Vector{typeof(transition)}(undef, N)
+    ts = Vector{typeof(transition)}(undef, 0)
+    Base.sizehint!(ts, N)
```

Comment:
It's exported, isn't it?
```suggestion
    sizehint!(ts, N)
```